File: Codigo/Nivel 1/Nivel 2/Nivel 3/Nivel 4/Ejemplos/activeMQ_python_ejemplo/MTISPublisher.py
import stomp

class MTISPublisher:
    def __init__(self, host='localhost', port=61613, topic='historiales'):
        # Nota: Si deseas usar el puerto 61616, cámbialo aquí (pero asegúrate de que el conector STOMP esté configurado para ese puerto)
        self.conn = stomp.Connection12(host_and_ports=[(host, port)])
        self.topic = topic

    def connect(self):
        self.conn.connect(wait=True)

    def send_message(self, message):
        destination = f"/topic/{self.topic}"
        self.conn.send(destination=destination, body=message)
        print(f"Sent message: {message}")

    def disconnect(self):
        self.conn.disconnect()

if __name__ == "__main__":
    publisher = MTISPublisher()
    publisher.connect()
    xml_message = "<history><patient><name>Manolo García</name></patient></history>"
    publisher.send_message(xml_message)
    publisher.disconnect()

File: Codigo/Nivel 1/Nivel 2/Nivel 3/Nivel 4/Ejemplos/activeMQ_javascript_ejemplo/MTISPublisher.js
// MTISPublisher.js

const stompit = require('stompit');

const connectOptions = {
  host: 'localhost',
  port: 61613,  // Nota: STOMP usa 61613 por defecto. Si deseas usar otro puerto, cámbialo aquí.
};

stompit.connect(connectOptions, (error, client) => {
    if (error) {
        console.error('Error de conexión: ' + error.message);
        return;
    }

    const destination = '/topic/historiales';
    const message = "<history><patient><name>Manolo García</name></patient></history>";

    // Enviar mensaje
    const frame = client.send({
        destination: destination,
        'content-type': 'text/plain'
    });
    frame.write(message);
    frame.end();

    console.log(`Mensaje enviado: ${message}`);

    // Se desconecta después de enviar el mensaje.
    client.disconnect();
});
